src/session.rs

```diff
@@ -1072,6 +1073,34 @@ impl Session {
         self.view_coords(self.views.active_id, p)
     }
 
+    /// Expand paths, e.g. by converting ~ into the user's home directory path
+    pub fn expand_path<P: AsRef<Path>>(&mut self, path: P) -> OsString {
+            let mut path = path.as_ref();
+
+            // for Linux and BSD and MacOS
+            if cfg!(unix) && path.starts_with("~") {
+
+                // ensure that the base directory list can be loaded by
+                // ensuring it has at least one element
+                if let Some(base_dirs) = dirs::BaseDirs::new() {
+
+                    // alloc space + store a copy of home directory as an
+                    // OsString, which are easily used with Path::new()
+                    let home = base_dirs.home_dir().as_os_str().to_os_string();
+
+                    if path == Path::new("~") {
+                        return home;
+                    }
+
+                    path = &path.strip_prefix("~").unwrap();
+                    return Path::new(&home).join(path).into_os_string();
+                }
+            }
+
+            // for Windows and other platforms
+            path.as_os_str().to_os_string()
```

Comment:
Is it necessary to return an `OsString`? Can we not return either `P` or simply `String`? If there's a good reason to return `OsString`, then perhaps we can make this a bit less quirky by saying `path.as_os_str().to_owned()`.

Reply:
Sure, I have since changed it to type String.